multy_core/src/EOS/EOS_transaction.cpp

```diff
@@ -0,0 +1,635 @@
+    result += ", ";
+    result += sig_base58;
+
+
+    return make_clone(as_binary_data(result));
+}
+
+
+void EOSTransaction::serialize_to_stream(EOSBinaryStream& stream, SerializationMode mode) const
+{
+    EOSBinaryStream list;
+    if (mode == SERIALIZE_FOR_SIGN)
+    {
+        list << EOS_TESTNET_CHAIN_ID;
+    }
+
+    list << static_cast<uint32_t>(m_expiration); // time sa uint32_t
+    list << static_cast<uint16_t>(static_cast<uint32_t>(*m_ref_block_num));
+    uint32_t ref_block_prefix;
+    if (*m_ref_block_prefix < std::numeric_limits<uint32_t>::max())
+    {
+        ref_block_prefix = static_cast<uint32_t>(m_ref_block_prefix.get_value().get_value_as_uint64());
+    }
+    else
+    {
+        THROW_EXCEPTION("Ref_block_prefix more larger uint32");
+    }
+    list << ref_block_prefix;
+    list << static_cast<uint8_t>(0x00); // max_net_usage_words, we set zero byte
+    list << static_cast<uint8_t>(0x00); // max_cpu_usage_ms, we set zero byte
+    list << static_cast<uint8_t>(0x00); // delay_seconds, we set zero byte
+    list << static_cast<uint8_t>(0x00); // Это надо переписывать использовав массив и сиреализуя сначала размер,
+                                        // а потом еще данные храняшиеся внутри массива
+
+    list << static_cast<uint8_t>(m_actions.size());
+    list << *m_actions.at(0);
+
+    list << static_cast<uint8_t>(0x00);
+
+    if (mode == SERIALIZE_FOR_SIGN)
+    {
+        list << EOS_ZERO_SHA256;
+    }
+
+    stream << list.get_content();
+}
+
+BigInt EOSTransaction::get_total_fee() const
+{
+    return BigInt(0);
+}
+
+BigInt EOSTransaction::get_total_spent() const
+{
+    if (!m_destination)
+    {
+        THROW_EXCEPTION("Failed to calculate total spent.")
+                << " Transaction has no destinations.";
+    }
+
+    return *m_destination->amount;
+}
+
+BigInt EOSTransaction::estimate_total_fee(
+        size_t /*sources_count*/,
+        size_t /*destinations_count*/) const
+{
+    return BigInt(0);
+}
+
+Properties& EOSTransaction::add_source()
+{
+    if (m_source)
+    {
+        THROW_EXCEPTION2(ERROR_TRANSACTION_TOO_MANY_SOURCES,
+                "EOS transaction can have only one source.");
+    }
+
+    m_source = EOSTransactionSourcePtr(new EOSTransactionSource(
+            get_blockchain_type()));
+
+    return m_source->get_properties();
+}
+
+Properties& EOSTransaction::add_destination()
+{
+    if (m_destination)
+    {
+        THROW_EXCEPTION2(ERROR_TRANSACTION_TOO_MANY_DESTINATIONS,
+                "EOS transaction can have only one destination.");
+    }
+
+    m_destination = EOSTransactionDestinationPtr(
+            new EOSTransactionDestination(get_blockchain_type()));
+
+    return m_destination->get_properties();
+}
+
+Properties& EOSTransaction::get_fee()
+{
+    THROW_EXCEPTION2(ERROR_FEATURE_NOT_SUPPORTED,
+            "EOS transaction fee is not customizable.");
+}
+
+void EOSTransaction::set_message(const BinaryData& value)
+{
```

Comment:
Validate that message is not longer than 255 bytes otherwise it wouldn't serialize correctly.
```
THROW_EXCEPTION2(ERROR_TRANSACTION_PAYLOAD_TO_BIG, "Message is to big.")
        << " Max message size is 255 bytes,"
        << " got: " << value.len() << " bytes.";
```